### ISSUE TITLE
clang-cl should provide implementations of more MSVC ARM64 intrinsics

### ISSUE THREAD

**Issue by rnk** [llvm/llvm-project#65405]
Some users have noticed that clang-cl does not provide some of the intrinsics listed in the [MSVC ARM64 intrinsic list](https://learn.microsoft.com/en-us/cpp/intrinsics/arm64-intrinsics?view=msvc-170). In particular, the `_Count*` and `_Copy*` intrinsics seem worth implementing, like `_CopyFloatFromInt32` and `_CountLeadingZeros`. See reference from the FP16 project [here](https://github.com/Maratyszcza/FP16/blob/0a92994d729ff76a58f692d3028ca1b64b145d91/include/fp16/bitcasts.h#L28).

cc @amykhuang @pzhengqc

**Comment by compnerd** [comment]
@EugeneZelenko I assume that the windows label removal was accidental (if not please re-remove and leave a note as to why the MSVC behaviours are not really windows-y).

**Comment by EugeneZelenko** [comment]
@compnerd: Isn't `clang-cl` Windows-only thing?

**Comment by compnerd** [comment]
@EugeneZelenko yes, it is a windows thing, but can be used on other platforms.  But that seems to indicate that it is supposed to be marked as windows (I was restoring the tag).

**Comment by davidbienvenu** [comment]
Another intrinsic that would be helpful is __prefetch.
